lib/waterline/query/aggregate.js

```diff
@@ -51,8 +53,49 @@ module.exports = {
       return value !== undefined;
     });
     
-    // Create will take care of cloning values so original isn't mutated
-    async.map(filteredValues, self.create.bind(self), cb);
+    // process each item, their associations and beforeCreate callbacks
```

Comment:
Shouldn't we do this **only** if the adapter implements createEach? Because otherwise the `.create` will take care of it, right? I may be wrong but I get the impression that for adapters that don't implement `createEach` the values are being processed twice...